pkg/parser/parser.go

```diff
@@ -2278,6 +2279,36 @@ func (p *Parser) parseAlterTable(pos token.Pos) *ast.AlterTable {
 	}
 }
 
+func (p *Parser) parseAlterTableAdd() ast.TableAlternation {
+	lexer := p.Lexer.Clone()
```

Comment:
Cloning lexer is a last resort for parsing. When it can be done without `Clone`, we recommend doing so.

@tyamagu2 If you have any idea that it must use `Clone`, please tell me or update a source code comment. If not, please consider implementing this without `Clone`. It seems not so hard.

Reply:
Maybe we can extract the following part of `parseAddColumn` as `parseColumn` and call it here?

https://github.com/MakeNowJust/memefish/pull/32/files#diff-2a6035c752559ce0fb74ced451e9e4363c01598c669d8f77a46be9450e6f6d87R2314-R2321